### DISCUSSION TITLE
Some Problems With The Pre-release version

### DISCUSSION THREAD

**Discussion by YinghuiYou** [microsoft/vscode-cpptools#8698] [Q&A]
After updating the extension to the Pre-Release  version, something strange came out. 
1. Variables and functions  are in white front, which can be seen from the picture below. 
![图片](https://user-images.githubusercontent.com/30487936/150339700-0fc1d505-3142-4d62-bcbf-c00f75e61d06.png)
2. The syntax doesn't work. In normal case, there ought to be a wave line under "EF".
![图片](https://user-images.githubusercontent.com/30487936/150340549-edb6746c-80a8-466f-95e1-b149959a9396.png)

I did no thing but updating the extension.  My computer operating system is Ubuntu20.04. 
I don't  know how to fix this problem. I have checked the settings of the extension, all the options are the same with before.

**Comment by sean-mcmanus** [comment]
Can you check the C/C++ logging for any "crash" messages? Do any extension features work?

**Reply by YinghuiYou** [reply]
I find no "crash" messages, and I can go to the definition by using the extension.

**Reply by sean-mcmanus** [reply]
The Go to Definition feature can work without the IntelliSense process (cpptools-srv).

It appears that our IntelliSense feature is somehow failing for you. Do you see any cpptools-srv processes running? Can you set C_Cpp.loggingLevel to Debug and then look in the C/C++ logs for any IntelliSense failure messages? Running the C/C++: Log Diagnostics command after opening the repro file would also confirm if the IntelliSense processes has started successfully or not.

You reported this 25 days ago when we had 1.8.0-insiders3. Which version are you experience this issue with now? Everything from 1.8.1 to 1.9.0? And only 1.7.1 works for you?

**Reply by YinghuiYou** [reply]
Thanks for your reply.

Rolling back the tool to a lower version like 1.7.1 or 1.2.2 didn't work. Updating to 1.9.0 remained the same result. It seems that I have met the same problem mentioned in [Issue#8790]( https://github.com/microsoft/vscode-cpptools/issues/8790)
Here is the Diagnostics Information.

> -------- Diagnostics - 2022/2/15下午5:17:28
> Version: 1.8.4
> Current Configuration:
> {
>     "browse": {
>         "limitSymbolsToIncludedHeaders": true,
>         "path": [
>             "/opt/ros/noetic/include/**",
>             "/home/yyh/REI_SLAM/src/reislam/include/**",
>             "/usr/include/**",
>             "/usr/lib/gcc/x86_64-linux-gnu/9/include/**",
>             "${workspaceFolder}"
>         ]
>     },
>     "includePath": [
>         "/opt/ros/noetic/include/**",
>         "/home/yyh/REI_SLAM/src/reislam/include/**",
>         "/usr/include/**",
>         "/usr/lib/gcc/x86_64-linux-gnu/9/include/**"
>     ],
>     "name": "ROS",
>     "intelliSenseMode": "gcc-x64",
>     "compileCommands": "${workspaceFolder}/build/compile_commands.json",
>     "compilerPathIsExplicit": false,
>     "cStandardIsExplicit": false,
>     "cppStandardIsExplicit": false,
>     "intelliSenseModeIsExplicit": true,
>     "compilerArgs": [],
>     "mergeConfigurations": false
> }
> Translation Unit Mappings:
> [ /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp ]:
>     /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
> Translation Unit Configurations:
> [ /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp ]:
>     Process ID: 28154
>     Memory Usage: 651 MB
>     Compiler Path: /usr/bin/c++
>     Includes:
>         /usr/include/eigen3
>         /opt/ros/noetic/include
>         /opt/ros/noetic/include/xmlrpcpp
>         /usr/include/opencv4
>         /usr/local/include/opencv
>         /usr/include/c++/9
>         /usr/include/x86_64-linux-gnu/c++/9
>         /usr/include/c++/9/backward
>         /usr/lib/gcc/x86_64-linux-gnu/9/include
>         /usr/local/include
>         /usr/include/x86_64-linux-gnu
>         /usr/include
>     Defines:
>         FMT_LOCALE
>         ROSCONSOLE_BACKEND_LOG4CXX
>         ROS_BUILD_SHARED_LIBS=1
>         ROS_PACKAGE_NAME="reislam_ros"
>     Standard Version: c++20
>     IntelliSense Mode: linux-gcc-x64
>     Other Flags:
>         --g++
>         --gnu_version=90300
>     compile_commands.json entry:
>         directory: /home/yyh/REI_SLAM/build/reislam
>         file: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
>         command: /usr/bin/c++  -DFMT_LOCALE -DROSCONSOLE_BACKEND_LOG4CXX -DROS_BUILD_SHARED_LIBS=1 -DROS_PACKAGE_NAME=\"reislam_ros\" -I/usr/include/eigen3 -I/opt/ros/noetic/include -I/opt/ros/noetic/share/xmlrpcpp/cmake/../../../include/xmlrpcpp -I/usr/include/opencv4 -isystem /usr/local/include/opencv   -Wall -Wunused-variable -Wpedantic -Werror -O3 -g -march=native  -fopenmp  -g -Wall -Wunused-variable -Wpedantic -Werror -O3 -g -march=native  -fopenmp    -std=gnu++2a -o CMakeFiles/reislam_live.dir/system/WindowedOptimizer.o -c /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
> Total Memory Usage: 651 MB
> Browse Paths from compile_commands.json, from workspace folder: /home/yyh/REI_SLAM
>     /home/yyh/REI_SLAM/src/reislam
>     /home/yyh/REI_SLAM/src/reislam/dataStructure
>     /home/yyh/REI_SLAM/src/reislam/factors
>     /home/yyh/REI_SLAM/src/reislam/system
>     /opt/ros/noetic/include
>     /opt/ros/noetic/include/xmlrpcpp
>     /usr/include/eigen3
>     /usr/include/opencv4
>     /usr/local/include/opencv
> 
> ------- Workspace parsing diagnostics -------
> Number of files discovered (not excluded): 37988
> 

I changed the C/C++ Log level to Debug, and reloaded the file. Here is the information.

> loggingLevel: Debug
> cpptools/didChangeCppProperties
> Attempting to get defaults from compiler found on the machine: '/usr/bin/gcc'
> Compiler query command line: "/usr/bin/gcc" -std=gnu17 -Wp,-v -E -dM -x c -m64  /dev/null
> terminating child process: 37217
> Code browsing service initialized
> Attempting to get defaults from compiler found on the machine: '/usr/bin/gcc'
> Compiler query command line: "/usr/bin/gcc" -std=gnu++14 -Wp,-v -E -dM -x c++ -m64  /dev/null
> terminating child process: 37230
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Querying compiler for default C++ language standard using command line: "/usr/bin/c++" -x c++ -E -dM /dev/null
> terminating child process: 37238
> Detected language standard version: gnu++14
> Querying compiler for default C language standard using command line: "/usr/bin/c++" -x c -E -dM /dev/null
> terminating child process: 37240
> Detected language standard version: gnu17
> Querying compiler's default target using command line: "/usr/bin/c++" -dumpmachine
> Compiler returned default target value: x86_64-linux-gnu
> 
> terminating child process: 37242
> Compiler query command line: "/usr/bin/c++" -Wall -Wunused-variable -Wpedantic -O3 -g -march=native -fopenmp -g -Wall -Wunused-variable -Wpedantic -O3 -g -march=native -fopenmp -std=gnu++2a  -Wp,-v -E -dM -x c++ -m64  /dev/null
> terminating child process: 37244
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
> Attempting to get defaults from C++ compiler in compile_commands.json file: '/usr/bin/c++'
>   Folder: /usr/include/ will be indexed
>   Folder: /usr/lib/gcc/x86_64-linux-gnu/9/include/ will be indexed
>   Folder: /usr/local/include/ will be indexed
>   Folder: /home/yyh/REI_SLAM/src/reislam/ will be indexed
>   Folder: /opt/ros/noetic/include/ will be indexed
> Discovering files...
> textDocument/didOpen: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
> cpptools/textEditorSelectionChange
> cpptools/activeDocumentChange: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
> sending compilation args for /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
>   include: /usr/include/eigen3
>   include: /opt/ros/noetic/include
>   include: /opt/ros/noetic/include/xmlrpcpp
>   include: /usr/include/opencv4
>   include: /usr/local/include/opencv
>   include: /usr/include/c++/9
>   include: /usr/include/x86_64-linux-gnu/c++/9
>   include: /usr/include/c++/9/backward
>   include: /usr/lib/gcc/x86_64-linux-gnu/9/include
>   include: /usr/local/include
>   include: /usr/include/x86_64-linux-gnu
>   include: /usr/include
>   define: __SSP_STRONG__=3
>   define: __DBL_MIN_EXP__=(-1021)
>   define: __FLT32X_MAX_EXP__=1024
>   define: __UINT_LEAST16_MAX__=0xffff
>   define: __ATOMIC_ACQUIRE=2
>   define: __FLT128_MAX_10_EXP__=4932
>   define: __FLT_MIN__=1.17549435082228750796873653722224568e-38F
>   define: __GCC_IEC_559_COMPLEX=2
>   define: __UINT_LEAST8_TYPE__=unsigned char
>   define: __SIZEOF_FLOAT80__=16
>   define: __INTMAX_C(c)=c ## L
>   define: __CHAR_BIT__=8
>   define: __UINT8_MAX__=0xff
>   define: __WINT_MAX__=0xffffffffU
>   define: __FLT32_MIN_EXP__=(-125)
>   define: __ORDER_LITTLE_ENDIAN__=1234
>   define: __SIZE_MAX__=0xffffffffffffffffUL
>   define: __SSE4_1__=1
>   define: __WCHAR_MAX__=0x7fffffff
>   define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1
>   define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1
>   define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1
>   define: __DBL_DENORM_MIN__=double(4.94065645841246544176568792868221372e-324L)
>   define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1
>   define: __GCC_ATOMIC_CHAR_LOCK_FREE=2
>   define: __GCC_IEC_559=2
>   define: __FLT32X_DECIMAL_DIG__=17
>   define: __znver2=1
>   define: __FLT_EVAL_METHOD__=0
>   define: __unix__=1
>   define: __FLT64_DECIMAL_DIG__=17
>   define: __CET__=3
>   define: __GCC_ATOMIC_CHAR32_T_LOCK_FREE=2
>   define: __x86_64=1
>   define: __RDPID__=1
>   define: __UINT_FAST64_MAX__=0xffffffffffffffffUL
>   define: __SIG_ATOMIC_TYPE__=int
>   define: __DBL_MIN_10_EXP__=(-307)
>   define: __FINITE_MATH_ONLY__=0
>   define: __GNUC_PATCHLEVEL__=0
>   define: __FLT32_HAS_DENORM__=1
>   define: __UINT_FAST8_MAX__=0xff
>   define: __FLT32_MAX_10_EXP__=38
>   define: __DEC64_MAX_EXP__=385
>   define: __INT8_C(c)=c
>   define: __INT_LEAST8_WIDTH__=8
>   define: __UINT_LEAST64_MAX__=0xffffffffffffffffUL
>   define: __SHRT_MAX__=0x7fff
>   define: __LDBL_MAX__=1.18973149535723176502126385303097021e+4932L
>   define: __FLT64X_MAX_10_EXP__=4932
>   define: __POPCNT__=1
>   define: __UINT_LEAST8_MAX__=0xff
>   define: __GCC_ATOMIC_BOOL_LOCK_FREE=2
>   define: __FLT128_DENORM_MIN__=6.47517511943802511092443895822764655e-4966F128
>   define: __UINTMAX_TYPE__=long unsigned int
>   define: __linux=1
>   define: __DEC32_EPSILON__=1E-6DF
>   define: __FLT_EVAL_METHOD_TS_18661_3__=0
>   define: __unix=1
>   define: __UINT32_MAX__=0xffffffffU
>   define: __GXX_EXPERIMENTAL_CXX0X__=1
>   define: __LDBL_MAX_EXP__=16384
>   define: __FLT128_MIN_EXP__=(-16381)
>   define: __WINT_MIN__=0U
>   define: __linux__=1
>   define: __FLT128_MIN_10_EXP__=(-4931)
>   define: __INT_LEAST16_WIDTH__=16
>   define: __SCHAR_MAX__=0x7f
>   define: __FLT128_MANT_DIG__=113
>   define: __WCHAR_MIN__=(-__WCHAR_MAX__ - 1)
>   define: __INT64_C(c)=c ## L
>   define: __DBL_DIG__=15
>   define: __GCC_ATOMIC_POINTER_LOCK_FREE=2
>   define: __FLT64X_MANT_DIG__=64
>   define: _FORTIFY_SOURCE=2
>   define: __SIZEOF_INT__=4
>   define: __SIZEOF_POINTER__=8
>   define: __GCC_ATOMIC_CHAR16_T_LOCK_FREE=2
>   define: __WBNOINVD__=1
>   define: __USER_LABEL_PREFIX__=
>   define: __FLT64X_EPSILON__=1.08420217248550443400745280086994171e-19F64x
>   define: __STDC_HOSTED__=1
>   define: __LDBL_HAS_INFINITY__=1
>   define: __FLT32_DIG__=6
>   define: __FLT_EPSILON__=1.19209289550781250000000000000000000e-7F
>   define: __GXX_WEAK__=1
>   define: __ABM__=1
>   define: __SHRT_WIDTH__=16
>   define: __LDBL_MIN__=3.36210314311209350626267781732175260e-4932L
>   define: __DEC32_MAX__=9.999999E96DF
>   define: __CHAR8_TYPE__=unsigned char
>   define: __F16C__=1
>   define: __FLT64X_DENORM_MIN__=3.64519953188247460252840593361941982e-4951F64x
>   define: __FLT32X_HAS_INFINITY__=1
>   define: __INT32_MAX__=0x7fffffff
>   define: __INT_WIDTH__=32
>   define: __SIZEOF_LONG__=8
>   define: __STDC_IEC_559__=1
>   define: __STDC_ISO_10646__=201706L
>   define: __UINT16_C(c)=c
>   define: __PTRDIFF_WIDTH__=64
>   define: __DECIMAL_DIG__=21
>   define: __FLT64_EPSILON__=2.22044604925031308084726333618164062e-16F64
>   define: __gnu_linux__=1
>   define: __INTMAX_WIDTH__=64
>   define: __FLT64_MIN_EXP__=(-1021)
>   define: __FLT64X_MIN_10_EXP__=(-4931)
>   define: __LDBL_HAS_QUIET_NAN__=1
>   define: __FLT64_MANT_DIG__=53
>   define: _REENTRANT=1
>   define: __GNUC__=9
>   define: __GXX_RTTI=1
>   define: __pie__=2
>   define: __MMX__=1
>   define: __XSAVES__=1
>   define: __FLT_HAS_DENORM__=1
>   define: __SIZEOF_LONG_DOUBLE__=16
>   define: __XSAVEOPT__=1
>   define: __BIGGEST_ALIGNMENT__=32
>   define: __STDC_UTF_16__=1
>   define: __PRFCHW__=1
>   define: __FLT64_MAX_10_EXP__=308
>   define: __FLT32_HAS_INFINITY__=1
>   define: __DBL_MAX__=double(1.79769313486231570814527423731704357e+308L)
>   define: __INT_FAST32_MAX__=0x7fffffffffffffffL
>   define: __DBL_HAS_INFINITY__=1
>   define: __SSE4_2__=1
>   define: __HAVE_SPECULATION_SAFE_VALUE=1
>   define: __DEC32_MIN_EXP__=(-94)
>   define: __INTPTR_WIDTH__=64
>   define: __FLT32X_HAS_DENORM__=1
>   define: __INT_FAST16_TYPE__=long int
>   define: __LDBL_HAS_DENORM__=1
>   define: __cplusplus=201709L
>   define: __DEC128_MAX__=9.999999999999999999999999999999999E6144DL
>   define: __INT_LEAST32_MAX__=0x7fffffff
>   define: __DEC32_MIN__=1E-95DF
>   define: __DEPRECATED=1
>   define: __DBL_MAX_EXP__=1024
>   define: __znver2__=1
>   define: __WCHAR_WIDTH__=32
>   define: __FLT32_MAX__=3.40282346638528859811704183484516925e+38F32
>   define: __DEC128_EPSILON__=1E-33DL
>   define: __SSE2_MATH__=1
>   define: __ATOMIC_HLE_RELEASE=131072
>   define: __PTRDIFF_MAX__=0x7fffffffffffffffL
>   define: __amd64=1
>   define: __AVX__=1
>   define: __ATOMIC_HLE_ACQUIRE=65536
>   define: __FLT32_HAS_QUIET_NAN__=1
>   define: __GNUG__=9
>   define: __LONG_LONG_MAX__=0x7fffffffffffffffLL
>   define: __SIZEOF_SIZE_T__=8
>   define: __LZCNT__=1
>   define: __FLT64X_MIN_EXP__=(-16381)
>   define: __SIZEOF_WINT_T__=4
>   define: __LONG_LONG_WIDTH__=64
>   define: __FLT32_MAX_EXP__=128
>   define: __GCC_HAVE_DWARF2_CFI_ASM=1
>   define: __GXX_ABI_VERSION=1013
>   define: __FLT128_HAS_INFINITY__=1
>   define: __FLT_MIN_EXP__=(-125)
>   define: __FLT64X_HAS_QUIET_NAN__=1
>   define: __INT_FAST64_TYPE__=long int
>   define: __FP_FAST_FMAF=1
>   define: __FLT64_DENORM_MIN__=4.94065645841246544176568792868221372e-324F64
>   define: __DBL_MIN__=double(2.22507385850720138309023271733240406e-308L)
>   define: __PIE__=2
>   define: __PCLMUL__=1
>   define: __LP64__=1
>   define: __FLT32X_EPSILON__=2.22044604925031308084726333618164062e-16F32x
>   define: __DECIMAL_BID_FORMAT__=1
>   define: __FLT64_MIN_10_EXP__=(-307)
>   define: __CLFLUSHOPT__=1
>   define: __FLT64X_DECIMAL_DIG__=21
>   define: __DEC128_MIN__=1E-6143DL
>   define: __REGISTER_PREFIX__=
>   define: __UINT16_MAX__=0xffff
>   define: __FLT32_MIN__=1.17549435082228750796873653722224568e-38F32
>   define: __UINT8_TYPE__=unsigned char
>   define: __XSAVE__=1
>   define: __CLWB__=1
>   define: __FLT_MANT_DIG__=24
>   define: __LDBL_DECIMAL_DIG__=21
>   define: __VERSION__="9.3.0"
>   define: __UINT64_C(c)=c ## UL
>   define: __FMA__=1
>   define: __SSE4A__=1
>   define: _STDC_PREDEF_H=1
>   define: __GCC_ATOMIC_INT_LOCK_FREE=2
>   define: __FLT128_MAX_EXP__=16384
>   define: __FLT32_MANT_DIG__=24
>   define: __FLOAT_WORD_ORDER__=__ORDER_LITTLE_ENDIAN__
>   define: __AVX2__=1
>   define: __STDC_IEC_559_COMPLEX__=1
>   define: __FLT128_HAS_DENORM__=1
>   define: __FLT128_DIG__=33
>   define: __SCHAR_WIDTH__=8
>   define: __INT32_C(c)=c
>   define: __DEC64_EPSILON__=1E-15DD
>   define: __ORDER_PDP_ENDIAN__=3412
>   define: __DEC128_MIN_EXP__=(-6142)
>   define: __INT_FAST32_TYPE__=long int
>   define: __UINT_LEAST16_TYPE__=short unsigned int
>   define: __FLT64X_HAS_INFINITY__=1
>   define: unix=1
>   define: __DBL_HAS_DENORM__=1
>   define: __INT16_MAX__=0x7fff
>   define: __SIZE_TYPE__=long unsigned int
>   define: __UINT64_MAX__=0xffffffffffffffffUL
>   define: __FLT64X_DIG__=18
>   define: __INT8_TYPE__=signed char
>   define: __ELF__=1
>   define: __GCC_ASM_FLAG_OUTPUTS__=1
>   define: __FLT_RADIX__=2
>   define: __INT_LEAST16_TYPE__=short int
>   define: __LDBL_EPSILON__=1.08420217248550443400745280086994171e-19L
>   define: __UINTMAX_C(c)=c ## UL
>   define: __GLIBCXX_BITSIZE_INT_N_0=128
>   define: __SIG_ATOMIC_MAX__=0x7fffffff
>   define: __GCC_ATOMIC_WCHAR_T_LOCK_FREE=2
>   define: __SIZEOF_PTRDIFF_T__=8
>   define: __FLT32X_MANT_DIG__=53
>   define: __OPTIMIZE__=1
>   define: __x86_64__=1
>   define: __FLT32X_MIN_EXP__=(-1021)
>   define: __DEC32_SUBNORMAL_MIN__=0.000001E-95DF
>   define: __INT_FAST16_MAX__=0x7fffffffffffffffL
>   define: __FLT64_DIG__=15
>   define: __UINT_FAST32_MAX__=0xffffffffffffffffUL
>   define: __UINT_LEAST64_TYPE__=long unsigned int
>   define: __FLT_HAS_QUIET_NAN__=1
>   define: __FLT_MAX_10_EXP__=38
>   define: __LONG_MAX__=0x7fffffffffffffffL
>   define: __FLT64X_HAS_DENORM__=1
>   define: __DEC128_SUBNORMAL_MIN__=0.000000000000000000000000000000001E-6143DL
>   define: __FLT_HAS_INFINITY__=1
>   define: __UINT_FAST16_TYPE__=long unsigned int
>   define: __DEC64_MAX__=9.999999999999999E384DD
>   define: __INT_FAST32_WIDTH__=64
>   define: __CHAR16_TYPE__=short unsigned int
>   define: __PRAGMA_REDEFINE_EXTNAME=1
>   define: __SIZE_WIDTH__=64
>   define: __SEG_FS=1
>   define: __INT_LEAST16_MAX__=0x7fff
>   define: __DEC64_MANT_DIG__=16
>   define: __INT64_MAX__=0x7fffffffffffffffL
>   define: __UINT_LEAST32_MAX__=0xffffffffU
>   define: __SEG_GS=1
>   define: __FLT32_DENORM_MIN__=1.40129846432481707092372958328991613e-45F32
>   define: __GCC_ATOMIC_LONG_LOCK_FREE=2
>   define: __SIG_ATOMIC_WIDTH__=32
>   define: __INT_LEAST64_TYPE__=long int
>   define: __INT16_TYPE__=short int
>   define: __INT_LEAST8_TYPE__=signed char
>   define: __DEC32_MAX_EXP__=97
>   define: __INT_FAST8_MAX__=0x7f
>   define: __FLT128_MAX__=1.18973149535723176508575932662800702e+4932F128
>   define: __INTPTR_MAX__=0x7fffffffffffffffL
>   define: linux=1
>   define: __FLT64_HAS_QUIET_NAN__=1
>   define: __FLT32_MIN_10_EXP__=(-37)
>   define: __SSE2__=1
>   define: __SSSE3__=1
>   define: __EXCEPTIONS=1
>   define: __RDRND__=1
>   define: __LDBL_MANT_DIG__=64
>   define: __DBL_HAS_QUIET_NAN__=1
>   define: __FLT64_HAS_INFINITY__=1
>   define: __FLT64X_MAX__=1.18973149535723176502126385303097021e+4932F64x
>   define: __STDCPP_DEFAULT_NEW_ALIGNMENT__=16
>   define: __SIG_ATOMIC_MIN__=(-__SIG_ATOMIC_MAX__ - 1)
>   define: __code_model_small__=1
>   define: __INTPTR_TYPE__=long int
>   define: __UINT16_TYPE__=short unsigned int
>   define: __WCHAR_TYPE__=int
>   define: __SIZEOF_FLOAT__=4
>   define: __pic__=2
>   define: __UINTPTR_MAX__=0xffffffffffffffffUL
>   define: __INT_FAST64_WIDTH__=64
>   define: __DEC64_MIN_EXP__=(-382)
>   define: __FLT32_DECIMAL_DIG__=9
>   define: __INT_FAST64_MAX__=0x7fffffffffffffffL
>   define: __GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1
>   define: __FLT_DIG__=6
>   define: __FLT64X_MAX_EXP__=16384
>   define: __UINT_FAST64_TYPE__=long unsigned int
>   define: __INT_MAX__=0x7fffffff
>   define: __amd64__=1
>   define: __SHA__=1
>   define: __GCC_ATOMIC_CHAR8_T_LOCK_FREE=2
>   define: __INT64_TYPE__=long int
>   define: __FLT_MAX_EXP__=128
>   define: __ORDER_BIG_ENDIAN__=4321
>   define: __DBL_MANT_DIG__=53
>   define: __SIZEOF_FLOAT128__=16
>   define: __INT_LEAST64_MAX__=0x7fffffffffffffffL
>   define: __FP_FAST_FMAF32=1
>   define: __DEC64_MIN__=1E-383DD
>   define: __WINT_TYPE__=unsigned int
>   define: __UINT_LEAST32_TYPE__=unsigned int
>   define: __SIZEOF_SHORT__=2
>   define: __SSE__=1
>   define: __LDBL_MIN_EXP__=(-16381)
>   define: __FLT64_MAX__=1.79769313486231570814527423731704357e+308F64
>   define: __WINT_WIDTH__=32
>   define: __FP_FAST_FMAF64=1
>   define: __INT_LEAST8_MAX__=0x7f
>   define: __FLT32X_MAX_10_EXP__=308
>   define: __SIZEOF_INT128__=16
>   define: __LDBL_MAX_10_EXP__=4932
>   define: __ATOMIC_RELAXED=0
>   define: __DBL_EPSILON__=double(2.22044604925031308084726333618164062e-16L)
>   define: __FLT128_MIN__=3.36210314311209350626267781732175260e-4932F128
>   define: _LP64=1
>   define: __UINT8_C(c)=c
>   define: __FLT64_MAX_EXP__=1024
>   define: __INT_LEAST32_TYPE__=int
>   define: __SIZEOF_WCHAR_T__=4
>   define: __FLT128_HAS_QUIET_NAN__=1
>   define: __INT_FAST8_TYPE__=signed char
>   define: __FLT64X_MIN__=3.36210314311209350626267781732175260e-4932F64x
>   define: __GNUC_STDC_INLINE__=1
>   define: __FLT64_HAS_DENORM__=1
>   define: _OPENMP=201511
>   define: __FLT32_EPSILON__=1.19209289550781250000000000000000000e-7F32
>   define: __FP_FAST_FMAF32x=1
>   define: __DBL_DECIMAL_DIG__=17
>   define: __STDC_UTF_32__=1
>   define: __INT_FAST8_WIDTH__=8
>   define: __FXSR__=1
>   define: __DEC_EVAL_METHOD__=2
>   define: __FLT32X_MAX__=1.79769313486231570814527423731704357e+308F32x
>   define: __UINT64_TYPE__=long unsigned int
>   define: __UINT32_C(c)=c ## U
>   define: __INTMAX_MAX__=0x7fffffffffffffffL
>   define: __BYTE_ORDER__=__ORDER_LITTLE_ENDIAN__
>   define: __FLT_DENORM_MIN__=1.40129846432481707092372958328991613e-45F
>   define: __INT8_MAX__=0x7f
>   define: __LONG_WIDTH__=64
>   define: __PIC__=2
>   define: __UINT_FAST32_TYPE__=long unsigned int
>   define: __CHAR32_TYPE__=unsigned int
>   define: __FLT_MAX__=3.40282346638528859811704183484516925e+38F
>   define: __FP_FAST_FMA=1
>   define: __INT32_TYPE__=int
>   define: __XSAVEC__=1
>   define: __SIZEOF_DOUBLE__=8
>   define: __FLT_MIN_10_EXP__=(-37)
>   define: __FLT64_MIN__=2.22507385850720138309023271733240406e-308F64
>   define: __INT_LEAST32_WIDTH__=32
>   define: __INTMAX_TYPE__=long int
>   define: __CLZERO__=1
>   define: __DEC128_MAX_EXP__=6145
>   define: __FSGSBASE__=1
>   define: __RDSEED__=1
>   define: __FLT32X_HAS_QUIET_NAN__=1
>   define: __ATOMIC_CONSUME=1
>   define: __GNUC_MINOR__=3
>   define: __GLIBCXX_TYPE_INT_N_0=__int128
>   define: __INT_FAST16_WIDTH__=64
>   define: __UINTMAX_MAX__=0xffffffffffffffffUL
>   define: __DEC32_MANT_DIG__=7
>   define: __FLT32X_DENORM_MIN__=4.94065645841246544176568792868221372e-324F32x
>   define: __DBL_MAX_10_EXP__=308
>   define: __LDBL_DENORM_MIN__=3.64519953188247460252840593361941982e-4951L
>   define: __BMI2__=1
>   define: __INT16_C(c)=c
>   define: __STDC__=1
>   define: __MWAITX__=1
>   define: __AES__=1
>   define: __FLT32X_DIG__=15
>   define: __PTRDIFF_TYPE__=long int
>   define: __ATOMIC_SEQ_CST=5
>   define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_16=1
>   define: __ADX__=1
>   define: __UINT32_TYPE__=unsigned int
>   define: __FLT32X_MIN_10_EXP__=(-307)
>   define: __UINTPTR_TYPE__=long unsigned int
>   define: __DEC64_SUBNORMAL_MIN__=0.000000000000001E-383DD
>   define: __DEC128_MANT_DIG__=34
>   define: __LDBL_MIN_10_EXP__=(-4931)
>   define: __FLT128_EPSILON__=1.92592994438723585305597794258492732e-34F128
>   define: __SSE_MATH__=1
>   define: __SIZEOF_LONG_LONG__=8
>   define: __FLT128_DECIMAL_DIG__=36
>   define: __GCC_ATOMIC_LLONG_LOCK_FREE=2
>   define: __FLT32X_MIN__=2.22507385850720138309023271733240406e-308F32x
>   define: __tune_znver2__=1
>   define: __LDBL_DIG__=18
>   define: __FLT_DECIMAL_DIG__=9
>   define: __UINT_FAST16_MAX__=0xffffffffffffffffUL
>   define: __GCC_ATOMIC_SHORT_LOCK_FREE=2
>   define: __INT_LEAST64_WIDTH__=64
>   define: __BMI__=1
>   define: __SSE3__=1
>   define: __UINT_FAST8_TYPE__=unsigned char
>   define: _GNU_SOURCE=1
>   define: __ATOMIC_ACQ_REL=4
>   define: __ATOMIC_RELEASE=3
>   define: FMT_LOCALE
>   define: ROSCONSOLE_BACKEND_LOG4CXX
>   define: ROS_BUILD_SHARED_LIBS=1
>   define: ROS_PACKAGE_NAME="reislam_ros"
>   other: --g++
>   other: --gnu_version=90300
>   stdver: c++20
>   intelliSenseMode: linux-gcc-x64
> cpptools/getCodeActions: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 2)
> cpptools/getFoldingRanges: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 3)
> cpptools/getSemanticTokens: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 4)
> cpptools/getCodeActions: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 5)
> cpptools/getDocumentSymbols: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 6)
> cpptools/getDocumentSymbols
> Checking for syntax errors: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
> Queueing IntelliSense update for files in translation unit of: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp
>   Processing folder (recursive): /usr/include/
>   Processing folder (recursive): /usr/lib/gcc/x86_64-linux-gnu/9/include/
>   Processing folder (recursive): /usr/local/include/
>   Processing folder (recursive): /home/yyh/REI_SLAM/src/reislam/
>   Processing folder (recursive): /opt/ros/noetic/include/
>   Discovering files: 37988 file(s) processed
>   0 file(s) removed from database
> Done discovering files.
> Parsing open files...
> Populating include completion cache.
> Done parsing open files.
> Parsing remaining files...
>   Parsing: 0 files(s) processed
> Done parsing remaining files.
> cpptools/finishUpdateSquiggles
> Error squiggle count: 0
> terminating child process: 37330
> Update IntelliSense time (sec): 9.637
> cpptools/getFoldingRanges: /home/yyh/REI_SLAM/src/reislam/system/WindowedOptimizer.cpp (id: 7)
>Quick info operation failed: FE: 'Unknown error'
>using Tag Parser for quick info
>cpptools/textEditorSelectionChange
>cpptools/textEditorSelectionChange

**Reply by sean-mcmanus** [reply]
You are hitting an "Unknown error" during hover which means our parser has entered an invalid state and has to abort. The issue https://github.com/microsoft/vscode-cpptools/issues/8790 is different -- it's a crash. Are you able to provide any more repro info?

**Reply by YinghuiYou** [reply]
Thanks for your answer. I'm afraid no more information can be provided.
